lib/services.js

```diff
@@ -585,48 +585,39 @@ export default {
     /**
      * Stores metadata about a part of a multipart upload
      * @param {string} mpuBucketName - name of the special mpu bucket
-     * @param {object} partLocations - data retrieval info for part
-     * @param {string} partLocations.key - key in datastore for part
-     * @param {string} partLocations.dataStoreName - name of dataStore
-     * @param {object} cipherBundle - server side encryption information
+     * @param {object []} partLocations - data retrieval info for part.
+     * @param {string} partLocations[].key -
+     * key in datastore for part
+     * @param {string} partLocations[].dataStoreName - name of dataStore
+     * @param {string} [partLocations[].size] - part size
+     * @param {string} [partLocations[].sseCryptoScheme] - cryptoScheme
+     * @param {string} [partLocations[].sseCipheredDataKey] - cipheredDataKey
+     * @param {string} [partLocations[].sseAlgorithm] - encryption algo
+     * @param {string} [partLocations[].masterKeyId] - masterKeyId
      * @param {object} metaStoreParams - custom built object
      * @param {object} log - request logger instance
      * @param {function} cb - callback to send error or move to next task
      * @return {undefined}
      */
     metadataStorePart(mpuBucketName, partLocations,
-                      cipherBundle, metaStoreParams, log, cb) {
+                      metaStoreParams, log, cb) {
         assert.strictEqual(typeof mpuBucketName, 'string');
-        const { partNumber, contentMD5, size, uploadId } = metaStoreParams;
-        const lastModified = new Date().toJSON();
-        const splitter = metaStoreParams.splitter;
+        const { partNumber, contentMD5, size, uploadId, lastModified, splitter }
+            = metaStoreParams;
+        const dateModified = typeof lastModified === 'string' ?
+            lastModified : new Date().toJSON();
         assert.strictEqual(typeof splitter, 'string');
-        // TODO: Determine splitter that will not appear in
-        // any of these items.  This is GH Issue#218
-        // 1) UploadId's are UUID version 4
-        // 2) Part Number will be a stringified number between 1 and 10000
         const partKey = `${uploadId}${splitter}${partNumber}`;
```

Comment:
This needs unification in the future (like a shared function) to make sure we generate the same format of key.